samoa-api/src/main/java/org/apache/samoa/evaluation/BasicClassificationPerformanceEvaluator.java

```diff
@@ -111,6 +125,38 @@ public void addResult(Instance inst, double[] classVotes) {
 
   }
 
+  /**
+   * This method is used to retrieve predictions and votes (for classification only)
+   * 
+   * @return String This returns an array of predictions and votes objects.
+   */
+  @Override
+  public Vote[] getPredictionVotes() {
```

Comment:
This has been modified. Assuming two classes in classification task (binary classification) there were ca. 5 objects created per every processed instance. Anyway, now these objects are reused. Similar solution exists for measurements.